src/Cassandra/Connections/Control/TopologyRefresher.cs

```diff
@@ -18,41 +18,54 @@
 using System.Linq;
 using System.Net;
 using System.Threading.Tasks;
+using Cassandra.Responses;
+using Cassandra.Tasks;
 
 namespace Cassandra.Connections.Control
 {
     /// <inheritdoc />
     internal class TopologyRefresher : ITopologyRefresher
     {
         private const string SelectPeers = "SELECT * FROM system.peers";
+        private const string SelectPeersV2 = "SELECT * FROM system.peers_v2";
         private const string SelectLocal = "SELECT * FROM system.local WHERE key='local'";
 
         private static readonly IPAddress BindAllAddress = new IPAddress(new byte[4]);
 
         private readonly Configuration _config;
         private readonly Metadata _metadata;
 
+        /// <summary>
+        /// Once this is set to false, it will never be set to true again.
+        /// </summary>
+        private volatile bool _isPeersV2 = true;
+
         public TopologyRefresher(Metadata metadata, Configuration config)
         {
             _metadata = metadata ?? throw new ArgumentNullException(nameof(metadata));
             _config = config ?? throw new ArgumentNullException(nameof(config));
         }
 
         /// <inheritdoc />
-        public async Task<Host> RefreshNodeListAsync(IConnectionEndPoint currentEndPoint, IConnection connection, ProtocolVersion version)
+        public async Task<Host> RefreshNodeListAsync(
+            IConnectionEndPoint currentEndPoint, IConnection connection, ProtocolVersion version)
         {
             ControlConnection.Logger.Info("Refreshing node list");
 
-            var queriesRs = await Task.WhenAll(
-                                          _config.MetadataRequestHandler.SendMetadataRequestAsync(
-                                              connection, version, TopologyRefresher.SelectLocal, QueryProtocolOptions.Default), 
-                                          _config.MetadataRequestHandler.SendMetadataRequestAsync(
-                                              connection, version, TopologyRefresher.SelectPeers, QueryProtocolOptions.Default))
-                                      .ConfigureAwait(false);
+            // safe guard against concurrent changes of this field
+            var localIsPeersV2 = _isPeersV2;
+
+            var localTask = SendSystemLocalRequestAsync(connection, version);
+            var peersTask = SendSystemPeersRequestAsync(localIsPeersV2, connection, version);
+            
+            await Task.WhenAll(localTask, peersTask).ConfigureAwait(false);
```

Comment:
NIT: There's no need to have `WhenAll()` as both tasks are awaited upon below.

Reply:
oh, true... my bad, nvm.

Reply:
Changed the following `await`s to `.Result`